### ISSUE TITLE
prefer-arrow-callback: allowUnboundThis description may be confusing

### ISSUE THREAD

**Issue by vsemozhetbyt** [eslint/eslint#8950]
http://eslint.org/docs/rules/prefer-arrow-callback#allowunboundthis

I've read this paragraph several times and by comparison with the code example it seems a bit confusing:

1. "When set to `false`, this option allows the use of `this` without restriction...": does not the option apply more restrictions for using `this`?

2. "...checks for dynamically assigned `this` values...": does "checks" mean "prohibits" or "allows"?

3. "Normally, the rule will flag...": does "Normally" mean "true" or "false" for this option?

The code example is clear, but the description seems to be vague.

**Comment by not-an-aardvark** [comment]
Thanks for reporting, I agree that the documentation is unclear/misleading here. To answer your questions:

* The default behavior is `{allowUnboundThis: true}`. With the default behavior, the rule allows function expressions containing `this`, provided that they are not explicitly bound. This is because these functions would not be directly replaceable with arrow functions.

    ```js
    foo(function() { this; }) // allowed by default
    foo(function() { this; }.bind(this)) // disallowed
    ```

When `allowUnboundThis` is set to `false`, the rule imposes a stricter requirement by disallowing all function expressions containing `this`, even though those functions are not directly replaceable with arrow functions.

**Comment by webdevdaemon** [comment]
Hello friends,

Looking to make my first ever PR on this one. Have signed the agreement and will follow the contribution guidelines to the best of my ability. If there is any objection to me taking a crack at this one - please advise.

Thanks for the opportunity,
-Charles

Also if anyone has the time, I am open to any suggestions or advice on the direction I should go in fixing this piece of the docs - but it seems pretty straightforward so I am just gonna go for it. I'll state again however, that any and all input is welcome.

**Comment by platinumazure** [comment]
@webdevdaemon Thanks for volunteering! We look forward to your PR :smile: I agree this should be pretty straightforward, so I would suggest giving it a try and then we can provide suggestions as needed while reviewing the pull request.

**Comment by webdevdaemon** [comment]
SWEET. Thx for the rapid reply - I'm on it.

**Comment by webdevdaemon** [comment]
I just realized that i PR'd 4 commits rather than just one... I'm sure you'll let me know if I need to fix that...

1000 apologies.

PR submitted though!
